src/xmidtsend_rbus.c

```diff
@@ -327,14 +359,70 @@ void sendXmidtEventToServer(wrp_msg_t * msg)
 		if(msg_len > 0)
 		{
 			ParodusInfo("sendUpstreamMsgToServer\n");
-			sendUpstreamMsgToServer(&msg_bytes, msg_len);
+			sendRetStatus = sendUpstreamMsgToServer(&msg_bytes, msg_len);
 		}
+		else
+		{
+			ParodusInfo("The msg_len is zero\n");
+			xmidtQDequeue();
+			wrp_free_struct(notif_wrp_msg);
+			ParodusInfo("After wrp_free_struct\n");
+			free(msg_bytes);
+			msg_bytes = NULL;
```

Comment:
Ack sending is required here since request processing is failed. Need to send callback for all other error cases such as producer/consumer enqueue failed . 
Add return types in each func and failure acks are not missed from each functions.